userspace/libpman/src/configuration.c

```diff
@@ -165,15 +169,126 @@ int pman_get_required_buffers()
 	return g_state.n_required_buffers;
 }
 
+static char byte_array[] = "BPF_TRACE_RAW_TP";
+
+bool check_BPF_TRACE_RAW_TP(void)
+{
+	// These locations are taken from libbpf library:
+	// https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4767
+	const char *locations[] = {
+		/* try canonical vmlinux BTF through sysfs first */
+		"/sys/kernel/btf/vmlinux",
+		/* fall back to trying to find vmlinux on disk otherwise */
+		"/boot/vmlinux-%1$s",
+		"/lib/modules/%1$s/vmlinux-%1$s",
+		"/lib/modules/%1$s/build/vmlinux",
+		"/usr/lib/modules/%1$s/kernel/vmlinux",
+		"/usr/lib/debug/boot/vmlinux-%1$s",
+		"/usr/lib/debug/boot/vmlinux-%1$s.debug",
+		"/usr/lib/debug/lib/modules/%1$s/vmlinux",
+	};
+	char path[PATH_MAX + 1];
+	struct utsname buf;
+	char *data = NULL;
+	FILE *f = NULL;
+	uname(&buf);
```

Comment:
I'd check the return code here.

Reply:
The reason why i didn't check it is that the first path to check is `"/sys/kernel/btf/vmlinux"`. Since it doesn't need the kernel release version even if `uname` fails probably we don't want to do anything and check at least for the first path  `"/sys/kernel/btf/vmlinux"`. In the worst case, when `uname` fails, we are just trying to read wrong paths, but `faccessat` should stop us. I think this is the reason why also in `libbpf` this function is not checked, see here https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4786

Reply:
Makes sense! Thanks!

Reply:
How about moving the inner part of the loop to a separate function, checking `/sys/kernel/btf/vmlinux` before calling `uname` and skipping the rest of the paths if `uname` fails?

Reply:
yeah it could be an idea, i'm not even sure if we want to keep the other paths, in container environment they would require extra volume mounts of `/boot` or `lib/modules` :/

Reply:
Well, we mount `/usr/lib` to `/host/usr/lib` in our use case, it'd bother me a bit if this started failing on some platform, I'd at least try to access them through `scap_get_host_root` as a fallback.

Reply:
Uhm if you mount `/usr/lib` under `/host/usr/lib` it means that libbpf won't be able to perform these checks in your container https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4786, is it true? or are you using some other workaround? This is in some way related to this https://github.com/falcosecurity/libs/pull/1404#discussion_r1355566414

Reply:
Solved here https://github.com/falcosecurity/libs/pull/1404#discussion_r1363566226